allennlp/data/tokenizers/word_tokenizer.py

```diff
@@ -38,42 +55,272 @@ class WordTokenizer(Tokenizer):
+        first check to be sure the token isn't in our list of special cases.
         """
-        words = self._word_splitter.split_words(text)
-        return self._filter_and_stem(words)
+        fields = text.split()
+        tokens: List[Token] = []
+        for field in fields:
+            add_at_end: List[Token] = []
+            while self._can_split(field) and field[0] in self.beginning_punctuation:
+                tokens.append(Token(field[0]))
+                field = field[1:]
+            while self._can_split(field) and field[-1] in self.ending_punctuation:
+                add_at_end.insert(0, Token(field[-1]))
+                field = field[:-1]
+
+            # There could (rarely) be several contractions in a word, but we check contractions
+            # sequentially, in a random order.  If we've removed one, we need to check again to be
+            # sure there aren't others.
+            remove_contractions = True
+            while remove_contractions:
+                remove_contractions = False
+                for contraction in self.contractions:
+                    if self._can_split(field) and field.lower().endswith(contraction):
+                        add_at_end.insert(0, Token(field[-len(contraction) :]))
+                        field = field[: -len(contraction)]
+                        remove_contractions = True
+            if field:
+                tokens.append(Token(field))
+            tokens.extend(add_at_end)
+        return tokens
+
+    def _can_split(self, token: str):
+        return token and token.lower() not in self.special_cases
+
+
+@Tokenizer.register("letters_digits")
+class LettersDigitsSpacyWordTokenizer(Tokenizer):
+    """
+    A ``Tokenizer`` which keeps runs of (unicode) letters and runs of digits together, while
+    every other non-whitespace character becomes a separate word.
+    """
+
+    @overrides
+    def tokenize(self, text: str) -> List[Token]:
+        # We use the [^\W\d_] pattern as a trick to match unicode letters
+        tokens = [Token(m.group(), idx=m.start()) for m in re.finditer(r"[^\W\d_]+|\d+|\S", text)]
+        return tokens
+
+
+@Tokenizer.register("just_spaces")
+class JustSpacesTokenizer(Tokenizer):
+    """
+    A ``Tokenizer`` that assumes you've already done your own tokenization somehow and have
+    separated the tokens by spaces.  We just split the input string on whitespace and return the
+    resulting list.  We use a somewhat odd name here to avoid coming too close to the more
+    commonly used ``SpacyTokenizer``.
+
+    Note that we use ``text.split()``, which means that the amount of whitespace between the
+    tokens does not matter.  This will never result in spaces being included as tokens.
+    """
+
+    @overrides
+    def tokenize(self, text: str) -> List[Token]:
+        return [Token(t) for t in text.split()]
+
+
+@Tokenizer.register("openai")
+class OpenAISpacyWordTokenizer(Tokenizer):
```

Comment:
`OpenAiTokenizer`.  We probably will want to just remove this later, in favor of the transformer tokenizer, but we can keep it for now, I suppose.